internal/tfengine/tfengine.go

```diff
@@ -80,8 +81,13 @@ func Run(confPath, outPath string, opts *Options) error {
 	return copy.Copy(tmpDir, outPath)
 }
 
-func dump(conf *Config, pwd, cacheDir, outputPath string) error {
+func dump(conf *Config, pwd, cacheDir, outputPath string, templates map[string]bool) error {
 	for _, ti := range conf.Templates {
+		// If a templates filter was provided, check against it.
+		if _, ok := templates[ti.Name]; len(templates) > 0 && !ok {
```

Comment:
optional: if you pop the key off the map then you can check for whether the list is non-empty at the end and detect the cases where an invalid key was passed in (e.g. to catch typos etc).

Reply:
Added. Opted not to pop the key, to avoid deleting the data and possibly causing issues in the future.

Reply:
You're still mutating the input so I would prefer popping as that would be simpler, but your call.

Reply:
Another option could have been to make a copy of the map and pass that in, though I am not sure if that makes things clearer.

Reply:
It gets reset back to the way it was at the end, so the function call is a no-op for that map.